tasty-query/shared/src/main/scala/tastyquery/reader/classfiles/JavaSignatures.scala

```diff
@@ -38,10 +38,51 @@ private[classfiles] object JavaSignatures:
       superRef :: interfaces.map(classRef).toList
   end parseSupers
 
+  def parseReturnDescriptor(descriptor: String)(using ReaderContext, InnerClasses, Resolver): Type =
```

Comment:
this is also implemented by the local method `result` inside of `parseSignature`, so I wonder if it can be reused with refactoring?

Reply:
Yes, I know. It's not that easy, however, because the more capable `parseSignature` expects a `member: TermOrTypeSymbol` that the signature attaches to, which we have no business having here. Factoring this out would require significant refactoring of the whole parsing, so I would rather not do that here (at least not at this point).